feat(storage): sample gap

AlisskaPie opened the pull request:
Samples to add with following region tags:
- [x] storage_bucket_delete_default_kms_key
- [ ] storage_generate_encryption_key - net
- [x] storage_view_bucket_iam_members
- [x] storage_add_bucket_label
- [ ] storage_compose_file
- [x] storage_remove_bucket_label
- [ ] storage_change_file_storage_class
- [ ] storage_enable_versioning
- [ ] storage_set_metadata
- [x] storage_set_bucket_public_iam
- [x] storage_define_bucket_website_configuration
- [ ] storage_download_public_file
- [ ] storage_get_service_account
- [x] storage_cors_configuration
- [ ] storage_disable_versioning
- [ ] storage_copy_file_archived_generation
- [ ] storage_object_csek_to_cmek
- [ ] storage_change_default_storage_class
- [ ] storage_delete_file_archived_generation
- [x] storage_remove_cors_configuration

There `storage_get_bucket_policy` region tag exist in the code, but not in Storage Sample Coverage - it's new `storage_view_bucket_iam_members`
P.S. to be consistent with other samples in document we now have old samples in main.go, but in new one separated samples need to change policy version to `V3`  also in `storage_add_bucket_iam_member` and `storage_remove_bucket_iam_member`. I changed version for renamed `storage_view_bucket_iam_members`
Version changed in:
- [x] storage_view_bucket_iam_members
- [ ] storage_add_bucket_iam_member
- [ ] storage_remove_bucket_iam_member

For [viewing the CORS configuration for a bucket](https://cloud.google.com/storage/docs/configuring-cors#viewing-cors-bucket) getBucketMetadata is used, but viewing for CORS configuration was missing.
 Added CORS configuration:

- [x] getBucketMetadata

AlisskaPie committed:
feat(storage): sample gap

AlisskaPie committed:
add_bucket_label

AlisskaPie committed:
add remove_bucket_label

AlisskaPie committed:
add storage_define_bucket_website_configuration

AlisskaPie committed:
add storage_set_bucket_public_iam

AlisskaPie committed:
fix main.go

AlisskaPie committed:
storage_view_bucket_iam_members: tag, use V3

AlisskaPie committed:
add storage_cors_configuration sample

AlisskaPie committed:
add storage_remove_cors_configuration sample

IlyaFaer commented on storage/buckets/add_bucket_label.go:
```diff
@@ -0,0 +1,52 @@
+
+	bucket := client.Bucket(bucketName)
+	bucketAttrsToUpdate := storage.BucketAttrsToUpdate{}
+	bucketAttrsToUpdate.SetLabel(labelName, labelValue)
```
Do we have an opportunity to set new label while calling `BucketAttrsToUpdate`? Without explicit calling `SetLabel`

AlisskaPie replied:
Good question, I've search for that opportunity, but it seems like we don't have it because labels in `BucketAttrsToUpdate` is [private](https://github.com/googleapis/google-cloud-go/blob/01ee6713322b0075df6dfad89b2031b6e4dbedaa/storage/bucket.go#L698). 
Another way to set [label](https://github.com/googleapis/google-cloud-go/blob/01ee6713322b0075df6dfad89b2031b6e4dbedaa/storage/bucket.go#L292) is to do it on `storage.BucketAttrs` level and to call `Bucket().Attrs()`, but this method is out of the general concept, I think. Also [example_test.go](https://github.com/googleapis/google-cloud-go/blob/01ee6713322b0075df6dfad89b2031b6e4dbedaa/storage/example_test.go#L133) uses `SetLabel`, that's why I decided to use it too

IlyaFaer commented on storage/buckets/set_bucket_public_iam.go:
```diff
@@ -0,0 +1,48 @@
+
+package buckets
+
+// [START storage_set_bucket_public_iam]
```
Aren't we already pushed such a sample? It sounds familiar

AlisskaPie replied:
Yeah, it's sounds familiar but it's not the same, because we're setting public IAM for a bucket only in this sample
We have following sample tags in the list and our sample looks pretty like [storage_add_bucket_iam_member](https://cloud.google.com/storage/docs/access-control/using-iam-permissions#bucket-add). But maybe we should change policy to `V3()` and don't pass policy straight?
```
storage_add_bucket_iam_member
storage_remove_bucket_iam_member
storage_view_bucket_iam_members
storage_set_bucket_public_iam
storage_add_bucket_conditional_iam_binding
storage_remove_bucket_conditional_iam_binding
```

IlyaFaer replied:
It sounds familiar to `storage_make_public`, I think, but they are not actually the same

AlisskaPie reviewed (commented):
Thanks for review

AlisskaPie committed:
add retry to addBucketLabel

AlisskaPie committed:
KMS, setBucketWebsiteInfo, BucketEncryption fixes

AlisskaPie committed:
test name TestBucketWebsiteInfo

AlisskaPie committed:
add checks for addBucketLabel, removeBucketLabel

AlisskaPie committed:
Merge branch 'master' into storage_sample_gap

AlisskaPie committed:
check kms key

AlisskaPie committed:
setBucketPublicIAM V3 policy, check, list buckets

AlisskaPie committed:
retry for removeBucketLabel

tritone committed:
Merge branch 'master' into storage_sample_gap

danoscarmike committed:
chore: updates region tags (#1790)

tbpg committed:
Merge branch 'master' into storage_sample_gap

tritone committed:
test(storage): add retry to s3sdk test (#1792)

This authentication error can occur as a transient error. Adding
a retry to see if this will fix the flakiness.

Fixes #1762

coryan committed:
fix(pubsub): fix incorrect comment (#1789)

Co-authored-by: Alex Hong <9397363+hongalex@users.noreply.github.com>

tritone committed:
Merge branch 'master' into storage_sample_gap
